HeterogeneousCore/CUDAUtilities/interface/AtomicPairCounter.h

```diff
@@ -0,0 +1,54 @@
+#ifndef HeterogeneousCoreCUDAUtilitiesAtomicPairCounter_H
+#define HeterogeneousCoreCUDAUtilitiesAtomicPairCounter_H
+
+#include<cstdint>
+
+class AtomicPairCounter {
+public:
+
+  using c_type = unsigned long long int;
+
+  AtomicPairCounter(){}
+  AtomicPairCounter(c_type i) { counter.ac=i;}
+
+#ifdef __CUDACC__
+  __device__ __host__
```

Comment:
if this is a `__host__ __device__` function, shouldn't it be defined also if we are not compiling CUDA code, i.e. if `__CUDACC__` is not defined ?

Reply:
this constructor yes.